harness-delegate-ng/values.yaml

```diff
@@ -109,6 +111,8 @@ nextGen: true
 accountId: ""
 # Delegate Token
 delegateToken: ""
+# Existing secret name with DELEGATE_TOKEN key
+existingDelegateToken: ""
```

Comment:
There's one case here that we would need to be sure of.
existingDelegateToken and existingUpgraderToken must not be created using helm delegate.

Consider case:
- We bring up delegate someDelegateName which adds secret someDelegateName
- We use existing secret someDelegateName to bringup new helm delegates.
- We do helm uninstall someDelegateName : This will delete the secret as well which is used by other delegate.
- Upgrade of all other delegates using deleted secret will fail with missing secret exception.
- cc: @mdmilic

Reply:
@AnupamIO , this case is similar to what @mdmilic mentioned before, and the answer is the same.

Using a secret managed by a helm chart in other helm deployments is a wrong pattern because the secret is part of the helm chart life-cycle and can be removed at any moment, but you cannot control it. This is a team responsibility.

Harness delegate helm chart should bring the option to use external secrets to enable well-known systems like Vault, 1Password, SealedSecrets, and others that manage the secrets in a different flow. With this feature, the chart offers the possibility to adopt the GitOps philosophy.

Reply:
Thanks @mmorejon for your contribution, the change looks good, however I do also agree with @AnupamIO about a problematic scenario. Unfortunately we often need to introduce guardrails to protect customers from themselves. 
I think the best approach here would be to just add some more descriptive doc comment in the `values.yaml` for the new variables explaining when they are OK to use and when not to use them. 
E.g. mention something along the lines that these should not be used with other helm managed, but only with externally managed secret resources (maybe mention Vault or other examples like you did in the comment above)

Reply:
New comments were added to those variables.
@mdmilic

Reply:
Thanks @mmorejon I fixed a typo and a minor clarification, but looks good to me. 
@AnupamIO can you please also double check and approve if it looks OK.